backend/project/routes_webauthn_duo.py

```diff
@@ -0,0 +1,282 @@
+
+    try:
+        webauthn_credential = webauthn_registration_response.verify()
+    except Exception as e:
+        return jsonify({'fail': 'Registration failed. Error: {}'.format(e)})
+
+    # Step 17.
+    #
+    # Check that the credentialId is not yet registered to any other user.
+    # If registration is requested for a credential that is already registered
+    # to a different user, the Relying Party SHOULD fail this registration
+    # ceremony, or it MAY decide to accept the registration, e.g. while deleting
+    # the older registration.
+    dbcursor.execute('SELECT credential_id FROM test WHERE credential_id = %s', (webauthn_credential.credential_id,))
+    credential_id_exists = dbcursor.fetchone()
+    
+    if credential_id_exists:
+        return make_response(
+            jsonify({
+                'fail': 'Credential ID already exists.'
+            }), 401)
+
+    dbcursor.execute('SELECT email FROM test WHERE email = %s', (email,))
+    existing_user = dbcursor.fetchone()
+    if not existing_user:
+        if sys.version_info >= (3, 0):
+            webauthn_credential.credential_id = str(
+                webauthn_credential.credential_id, "utf-8")
+            webauthn_credential.public_key = str(
+                webauthn_credential.public_key, "utf-8")
+        '''
+        user = User(
+            ukey=ukey,
+            username=username,
+            display_name=display_name,
+            pub_key=webauthn_credential.public_key,
+            credential_id=webauthn_credential.credential_id,
+            sign_count=webauthn_credential.sign_count,
+            rp_id=RP_ID,
+            icon_url='https://example.com')
+        db.session.add(user)
+        db.session.commit()
+        '''
+        # Add new user entry to database table
+        dbcursor.execute('INSERT INTO users (email, display_name, ukey, icon_url, credential_id, pub_key, sign_count, rp_id) VALUES (%s, %s, %s, %s, %s, %s, %d, %s)', (email, display_name, ukey, 'https://example.com', webauthn_credential.credential_id, webauthn_credential.public_key, webauthn_credential.sign_count, RP_ID))
+        dbconnection.commit()
+    else:
+        return make_response(jsonify({'fail': 'User already exists.'}), 401)
+
+    #flash('Successfully registered with email {}.'.format(email))
+
+    return jsonify({'success': 'User successfully registered.'})
+
+
+@app.route('/verify_assertion', methods=['POST'])
+def verify_assertion():
+    challenge = session.get('challenge')
+    assertion_response = request.form
+    credential_id = assertion_response.get('id')
+
+    dbcursor.execute('SELECT email, display_name, ukey, icon_url, credential_id, pub_key, sign_count, rp_id FROM test WHERE credential_id = %s', (credential_id,))
+    user = dbcursor.fetchone()
+    if not user:
+        return make_response(jsonify({'fail': 'User does not exist.'}), 401)
+
+    webauthn_user = webauthn.WebAuthnUser(
+        user[2], # user.ukey
```

Comment:
There's a super-cool feature called named tuples, which allow you to access particular tuple fields by key name. I would recommend setting this up as indicated here: https://overiq.com/mysql-connector-python-101/executing-queries-using-connector-python/#mysqlcursornamedtuple-class.

Then, you could just access this as `user.ukey`, `user.username`, etc.